server/server.go

```diff
@@ -110,9 +110,24 @@ func (s *Server) serve(listener net.Listener, readyCb func(addr string)) error {
 		readyCb(l.Addr().String())
 	}
 
+	var tempDelay time.Duration // how long to sleep on accept failure
 	for {
 		conn, err := l.Accept()
 		if err != nil {
+			if ne, ok := err.(net.Error); ok && ne.Temporary() {
```

Comment:
Previously, any Accept error made us terminate the http-proxy. Now, if we detect temporary network errors, we wait and retry.